spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/DefaultTokenServices.java

```diff
@@ -211,6 +207,32 @@ private OAuth2Authentication createRefreshedAuthentication(OAuth2Authentication
 		return narrowed;
 	}
 
+	/**
+	 * Re-authenticate the user authentication to verify it is up-to-date.
+	 * @param authentication The authentication
+	 * @return The re-authenticated authentication.
+	 */
+	protected OAuth2Authentication reauthenticateUserAuthentication(OAuth2Authentication authentication) {
```

Comment:
Please remove `reauthenticateUserAuthentication()` and `createUserPreAuthenticatedToken()` as it's not needed. The only change required is to `setDetails()` on the `PreAuthenticatedAuthenticationToken` before it's passed to `AuthenticationManager`.

Reply:
> Thanks for the PR @klieber. If you're still looking to get this merged, please see my review comment and apply the change and we'll get this merged.

Sure I will take a look.

Reply:
@jgrandja I've made the changes requested.  Let me know if there is anything else needed.